--- pyconfig.h	Sun Jun 29 22:05:46 2014
+++ pyconfig_nodebug.h	Sat Nov 22 01:55:51 2014
@@ -323,7 +323,7 @@
 			their Makefile (other compilers are generally
 			taken care of by distutils.) */
 #			ifdef _DEBUG
-#				pragma comment(lib,"python27_d.lib")
+#				pragma comment(lib,"python27.lib")
 #			else
 #				pragma comment(lib,"python27.lib")
 #			endif /* _DEBUG */
@@ -365,7 +365,7 @@
 #endif
 
 #ifdef _DEBUG
-#	define Py_DEBUG
+//#	define Py_DEBUG
 #endif
 
 
